src/video_stream.cpp

```diff
@@ -137,11 +137,12 @@ virtual void do_capture() {
             camera_fps_rate.sleep();
         }
         if (video_stream_provider_type == "videofile" &&
-            frame_counter == cap->get(cv::CAP_PROP_FRAME_COUNT))
+            frame_counter == latest_config.stop_frame - latest_config.start_frame)
         {
             if (latest_config.loop_videofile)
             {
                 cap->open(video_stream_provider);
+                cap->set(cv::CAP_PROP_POS_FRAMES, latest_config.start_frame);
```

Comment:
It's better to check if the start_frame doesn't exceed the length of a video frame as same as stop_frame.

Reply:
@furushchev 

Thank you for your kind review!
Please check [this commit](https://github.com/ros-drivers/video_stream_opencv/pull/57/commits/34dadf9f8a5507779c7786263c8f1d2e18ececdf).